docs/mvvm/MigratingFromMvvmLight.md

```diff
@@ -0,0 +1,842 @@
+{
+    services.AddSingleton<IDialogService, DialogService>();
+    services.AddSingleton<IToastService, ToastService>();
+    services.AddSingleton<INavigationService, NavigationService>();
+});
+```
+
+## Migrating Messenger
+
+The following steps focus on migrating your existing components which take advantage of the `Messenger` of the MvvmLight Toolkit. 
+
+The Windows Community Toolkit MVVM framework provides a [`Messenger`](Messenger) type that provides similar functionality. 
+
+A note on `Register` methods throughout this migration. MvvmLight uses weak references to establish the link between the `Messenger` instance and the recipient. This is not required by the MVVM Toolkit implementation and if this optional parameter has been set to `true` in any of your `Register` method calls, this will be removed.
+
+Below are a list of migrations that will need to be performed if being used in your current solution.
+
+The first change here will be swapping using directives in your components.
+
+```csharp
+// MvvmLight
+using GalaSoft.MvvmLight.Messaging;
+
+// Toolkit.Mvvm
+using Microsoft.Toolkit.Mvvm.Messaging;
+```
+
+### Messenger Methods
+
+#### Register<TMessage> ( object, Action<TMessage> )
+
+The functionality of `Register<TMessage>(object, Action<TMessage>)` can be achieved with the MVVM Toolkit's `IMessenger` extension method `Register<TMessage>(object, Action<TMessage>)`. 
+
+```csharp
+// MvvmLight
+Messenger.Default.Register<MyMessage>(this, this.OnMyMessageReceived);
+
+// Toolkit.Mvvm
+Messenger.Default.Register<MyMessage>(this, this.OnMyMessageReceived);
+```
+
+#### Register<TMessage> ( object, bool, Action<TMessage> )
+
+There is no direct replacement for this registration mechanism which allows you to support receiving messages for derived message types also. This change is intentional as the `Messenger` implementation aims to not use reflection to achieve its performance benefits.
+
+Alternatively, there are a few options that can be done to achieve this functionality. 
+
+- Create a custom `IMessenger` implementation
+- Register the additional message types using a shared `Action`
+
+```csharp
+// MvvmLight
+Messenger.Default.Register<MyMessage>(this, true, this.OnMyMessageReceived);
+
+// Toolkit.Mvvm
+Messenger.Default.Register<MyMessage, string>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+Messenger.Default.Register<MyOtherMessage, string>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+```
+
+#### Register<TMessage> ( object, object, Action<TMessage> )
+
+The functionality of `Register<TMessage>(object, object, Action<TMessage>)` can be achieved with the MVVM Toolkit's `Register<TMessage, TToken>(object, TToken, Action<TMessage>)` method. 
+
+```csharp
+// MvvmLight
+Messenger.Default.Register<MyMessage>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+
+// Toolkit.Mvvm
+Messenger.Default.Register<MyMessage, string>(this, nameof(MyViewModel), this.OnMyMessageReceived);
```

Comment:
Same as above, this would be:

```csharp
Messenger.Default.Register<MyViewModel, MyMessage, string>(this, (r, m) => r.OnMyMessageReceived(m), nameof(MyViewModel));
```

Also I think we should call out the optional registration mode using the `IRecipient<T>` interface? As that's much less verbose to wire up compared to this, and could be a nice new feature for some devs migrating 😄